client/core/bond.go

```diff
@@ -39,58 +40,217 @@ func (c *Core) watchExpiredBonds(ctx context.Context) {
+//
+//   t0  t1                      t2'    t2                   t3  t4
+//   |~~~|-----------------------^------|====================|+++|
+//     ~                 -                        =
+//  pending (p)       live (l)                expired (E)   maturing (m)
+//
+//    t0 = authoring/broadcast
+//    t1 = activation (confirmed and accepted)
+//    t2 = expiry (tier accounting)
+//    t3 = lockTime
+//    t4 = spendable (block medianTime > lockTime)
+//
+//    E = t3 - t2, *constant* duration, dex.BondExpiry()
+//    p = t1 - t0, *variable*, a random process
+//    m = t4 - t3, *variable*, depends on consensus rules and blocks
+//        e.g. several blocks after lockTime passes
+//
+//  - bonds may be spent at t4
+//  - bonds must be replaced by t2 i.e. broadcast by some t2'
+//  - perfectly aligning t4 for bond A with t2' for bond B is impossible on
+//    account of the variable durations
+//  - t2-t2' should be greater than a large percent of expected pending
+//    durations (see pendingBuffer)
+//
+// Here a replacement bond B had a long pending period, and it became active
+// after bond A expired (too late):
+//
+//             t0  t1                         t2' t2                   t3
+//   bond A:   |~~~|--------------------------^---|====================|
+//                                                x
+//   bond B:                                  |~~~~~~|------------------...
+//
+// Here the replacement bond was confirmed quickly, but l was too short,
+// causing it to expire before bond A became spendable:
+//                                                                        > renew as bond C
+//   bond A:   |~~~|----------------------^-------|====================|++‖~~~~~|---...
+//                                          ✓                             x     x
+//   bond B:                              |~|------------------------|=====...
+//
+// Similarly, l could have been long enough to broadcast a replacement in time,
+// but the pending period could be too long (the second "x").
+//
+// Here the replacement bond was broadcast with enough time to confirm before
+// the previous bond expired, and the previous bond became spendable in time to
+// broadcast and confirm another replacement (sustainable):
+//                                                                        > renew as bond C
+//   bond A:   |~~~|----------------------^-------|====================|++‖~~~~~|---...
+//                                               ✓                        ✓     ✓
+//   bond B:                              |~~~~~~|-------------------------------|====...
+//
+// Thus, bond rotation without tier drops requires l>E+m+p. For
+// t3-t0 = p+l+E, this means t3-t0 >= 2(E+p)+m. We will assume the time
+// from authoring to broadcast is negligible, absorbed into the estimate of the
+// max pending duration.
+//
+// tldr:
+//  - when creating a bond, set lockTime = 2*(BondExpiry+pendingBuffer)+spendableDelay
```

Comment:
Good points.  I was actually considering writing these as `lockTime - t0 = ...` but didn't want them even longer, but since you are picking up on the correctness of that, I will update.